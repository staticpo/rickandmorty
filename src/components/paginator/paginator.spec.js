import React from 'react';
import { shallow, render, mount } from 'enzyme';
import escapeSnapshot from '../../../tests/commons';
import Paginator from './index';

describe('Profile (Snapshot)', () => {
  const defaultProps = {
    currentPage: 0,
    pages: 20,
    next: 1,
    prev: undefined,
    eventHandler: jest.fn(() => {}),
  };

  it('renders in page 0', () => {
    const component = shallow(<Paginator {...defaultProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('renders in page 10', () => {
    defaultProps.currentPage = 10;
    defaultProps.prev = 9;
    const component = shallow(<Paginator {...defaultProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
