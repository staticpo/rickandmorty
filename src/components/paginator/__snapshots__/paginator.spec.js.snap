// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile (Snapshot) renders in page 0 1`] = `
"<div className='paginator'>
  <button className='paginator__button' onClick={[Function: onClick]}>
    1
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    2
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    3
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    4
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    5
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    6
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    Next
  </button>
</div>"
`;

exports[`Profile (Snapshot) renders in page 10 1`] = `
"<div className='paginator'>
  <button className='paginator__button' onClick={[Function: onClick]}>
    Prev
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    5
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    6
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    7
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    8
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    9
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    10
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    11
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    12
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    13
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    14
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    15
  </button>
  <button className='paginator__button' onClick={[Function: onClick]}>
    Next
  </button>
</div>"
`;
